pkg/controller/controllercmd/builder.go

```diff
@@ -95,6 +95,9 @@ type ControllerBuilder struct {
 	// We use this flag to determine at runtime if we can alter leader election for SNO configurations
 	userExplicitlySetLeaderElectionValues bool
 
+	// different deployment strategies will require sensing topologies in disjoint manners
```

Comment:
make the interface broader to allow people to avoid using restClients which resist unit testing.